### PR TITLE
Remove Apache dependency

### PR DESCRIPTION
**Pull request by antslava**
Remove Apache dependency and add workaround for Robolectric issue robolectric/robolectric#1862 until Robolectric release 3.1

### CODE REVIEW

**Comment by itstexter** [comment]
I was hesitant about that workaround before because I was not certain that I wasn't using AndroidHttpClient but if all the tests pass and it looks like it is working I'm fine with it.

File: sdk/src/test/java/android/net/http/AndroidHttpClient.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package android.net.http;
```
Comment (itstexter) [inline sdk/src/test/java/android/net/http/AndroidHttpClient.java]: Add Uber License above package

File: sdk/src/test/java/android/net/http/AndroidHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+/**
+ * Created by Antonenko Viacheslav on 24/02/16.
```
Comment (itstexter) [inline sdk/src/test/java/android/net/http/AndroidHttpClient.java]: Note why this class is necessary and possibly link to robolectric issue thread?

**Comment by antslava** [comment]
@itstexter check it please.

File: sdk/src/test/java/android/net/http/AndroidHttpClient.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+package android.net.http;
+
+/**
+ * Since Apache HTTP Client was deprecated, it was no longer on the compile classpath. 
+ * However, Robolectric 3.0 still links to it. So we need to add no-op class for successful compiling.
+ * This workaround work only if we do not use AndroidHttpClient in our test cases. 
+ * 
+ * Link to Robolectric issue https://github.com/robolectric/robolectric/issues/1862
+ * 
+ * Created by Antonenko Viacheslav on 24/02/16.
```
Comment (itstexter) [inline sdk/src/test/java/android/net/http/AndroidHttpClient.java]: Can you remove this line?  Sorry, we don't have any attribution in our JavaDocs.

**Comment by itstexter** [comment]
@antslava Sorry for latency on this, added one more comment about JavaDoc. I can actually merge this pull request after that though!

**Comment by tyvsmith** [comment]
No longer relevant to current codebase.

**Pull request closed without merging**